src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -305,9 +307,11 @@ private void WatchForFileSystemEventsThreadStart(object arg)
 
                 for (long i = 0; i < numEvents.ToInt32(); i++)
                 {
+                    Debug.Assert(eventPaths[i][eventPaths[i].Length - 1] != '/');
+
                     // Match Windows and don't notify us about changes to the Root folder
                     string path = eventPaths[i];
-                    if (path.Equals(_fullDirectory, StringComparison.OrdinalIgnoreCase))
+                    if (string.Compare(path, 0, _fullDirectory, 0, path.Length) == 0)
```

Comment:
Compare isn't doing the same OrdinalIgnoreCase comparison that was previously being done with Equals.  This overload of Compare does culture-specific comparison based on the current culture, and is case-sensitive.  Seems like we're then also missing tests?